### ISSUE TITLE
[Bug]: Missing robots.txt when using custom domain (cname)

### ISSUE THREAD

**Issue by dsiscout** [zitadel/zitadel#5632]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

ZITADEL Cloud

### Describe the bug

We have configured a CNAME for our production environment. Unfortunately due to missing robots.txt, google is indexing our login page, which should not be the case. Attached you will find a screenshot of the google search result.

### To reproduce

1. Create an environment with CNAME configured
2. Integrate it with a frontend application
3. Google is indexing your page + login page

### Screenshots

![image](https://user-images.githubusercontent.com/55233123/230583775-6d347658-c7a8-4f54-892a-99c40a844676.png)

### Expected behavior

Search engines should not be permitted to index the login page. This can be achieved by configuring the robots.txt file or sending the `X-Robots-Tag: noindex` header.

### Version

Cloud

### Operating System

irrelevant

**Comment by doncicuto** [comment]
@hifabienne, maybe a duplicate of #5361? You can assign that issue to me if you want. I'll work on it in parallel with my current assignments

**Comment by hifabienne** [comment]
Hei @doncicuto 
Yes I think it is a duplicate.
Thanks for contributing.

**Comment by hifabienne** [comment]
Yes. Thanks for the mention.
